src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py

```diff
@@ -181,6 +181,11 @@ def launch(
                 self.name,
             )
 
+        settings = cast(SagemakerStepOperatorSettings, self.get_settings(info))
+
+        if settings.environment:
+            environment.update(settings.environment)
```

Comment:
This would allow users to potentially overwrite some crucial environment variables that we need to set in order to run steps

Reply:
I think we should let them :smile:

Reply:
I see this rather as a feature than a bug. It allows users to hack ZenML in all sorts of twisted ways while waiting for official releases to deliver their bug fixes / features.